cockatrice/src/player.cpp

```diff
@@ -1126,46 +1126,175 @@ void Player::actCreateRelatedCard()
     CardItem * sourceCard = game->getActiveCard();
     if(!sourceCard)
         return;
-
     QAction *action = static_cast<QAction *>(sender());
-    const QString &actionDisplayName = action->text();
-    createCard(sourceCard, dbNameFromTokenDisplayName(actionDisplayName));
+    //If there is a better way of passing a CardRelation through a QAction, please add it here.
+    QList<CardRelation *> relatedCards = QList<CardRelation *>();
+    relatedCards.append(sourceCard->getInfo()->getRelatedCards());
+    relatedCards.append(sourceCard->getInfo()->getReverseRelatedCards2Me());
+    CardRelation * cardRelation = relatedCards.at(action->data().toInt());
+
```

Comment:
> //If there is a better way of passing a CardRelation through a QAction, please add it here.

Ah, haha. I was actually hoping there was something I was missing, a better way of doing it.
Oh dear. :poop: